### PR TITLE
Identity Provider Interface Revamped in the Guardio Insurance Sample Application

### PR DESCRIPTION
**Pull request by Achintha444**
## User stories

>#### Story 01: Add an identity provider to the Identity Server.
>#### Story 02: Edit an identity provider to the Identity Server.
>#### Story 03: Delete an identity provider to the Identity Server.

### CODE REVIEW

**Comment by Achintha444** [comment]
### Related Issues

- https://github.com/wso2-enterprise/iam-engineering/issues/65
- https://github.com/wso2-enterprise/asgardeo-product/issues/12530

**Comment by Achintha444** [comment]
https://github.com/wso2-enterprise/iam-engineering/issues/65

File: b2b-sample/guardio-insurance-sample-application/components/data/templates/enterprise-identity-provider.json [hunk 1]
```diff
@@ -36,4 +109,4 @@
   "disabled": false,
   "templateId": "enterprise-idp",
   "type": "ENTERPRISE"
-}
+}
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/data/templates/enterprise-identity-provider.json:112]: Add an empty line here

File: b2b-sample/guardio-insurance-sample-application/components/data/templates/google.json [hunk 2]
```diff
@@ -85,12 +65,41 @@
...
         }
       ]
-    }
+    },
+    "templateId": "google-idp"
   },
   "type": "SOCIAL",
   "templateId": "google-idp"
-}
+}
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/data/templates/google.json:105]: Add an empty line.

File: b2b-sample/guardio-insurance-sample-application/components/editUser/editUserComponent.js [hunk 3]
```diff
@@ -175,7 +176,10 @@ export default function EditUserComponent(props) {
                                     <FormSuite.Group>
                                         <ButtonToolbar>
                                             <Button className={styles.addUserButton} size="lg" appearance="primary"
-                                                type='submit' disabled={submitting || pristine}>Submit</Button>
+                                                type='submit'
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/editUser/editUserComponent.js:179]: Shall we use double quotes?

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js [hunk 4]
```diff
@@ -115,131 +114,150 @@ export default function AddUserComponent(props) {
...
-                        </FormSuite>
-                    )}
-                />
-            </div>
+        <Modal backdrop="static" role="alertdialog" open={props.open} onClose={props.onClose} >
+
+            <Modal.Header>
+                <Modal.Title>
+                    <b>Add User</b>
+                    <p>Add a new user</p>
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js:122]: Can we be consistent with the case here?

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identityProviderDetails.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import React, { useCallback, useEffect, useState } from 'react';
+import { Avatar, Nav, Panel, Stack, useToaster } from 'rsuite';
+import decodeGetDetailedIdentityProvider from
+  '../../../util/apiDecode/settings/identityProvider/decodeGetDetailedIdentityProvider';
+import ButtonGroupIdentityProviderDetails from './buttonGroupIdentityProviderDetails';
+import General from './idpDetailsSections/general';
+import Raw from './idpDetailsSections/raw';
+import Settings from './idpDetailsSections/settings';
+
+export default function IdentityProviderDetails(props) {
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identityProviderDetails.js:28]: We should be using four spaces for indentation.

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/general.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+    const [loadingDisplay, setLoadingDisplay] = useState(LOADING_DISPLAY_NONE);
+
+    const toaster = useToaster();
+
+    const nameValidate = (name, errors) => {
+        if (!name) {
+            errors.name = 'This field cannot be empty'
+        }
+        return errors;
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/general.js:42]: Leave an empty line before the return statement. Check other places as well.

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callPatchGeneralSettingsIdp.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+         );
+ 
+         const data = await res.json();
+ 
+         return data;
+     } catch (err) {
+         return null;
+     }
+ }
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callPatchGeneralSettingsIdp.js:40]: Add new empty line

File: b2b-sample/guardio-insurance-sample-application/util/news/news.json [hunk 8]
```diff
@@ -1,24 +1,28 @@
...
         },
         {
+            "id": 4,
             "header": "Noyota Will Buy Back bZ4X EVs over Wheel Issue",
             "body": "When Noyota first announced a recall for its new, electric bZ4X SUV, it said that the hub bolts used on the wheels might loosen to the point where the wheel can detach from the vehicle. That was a noticeable black eye for one of the world’s largest automakers launching a crucial new model.",
             "image": "/../../../public/news1.jpeg"
         }
     ]
-}
+}
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/news/news.json:28]: Add new empty line

File: b2b-sample/guardio-insurance-sample-application/util/util/idpUtil/idpUtil.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+            "value": "scope=email openid profile"
+        }
+    ];
+
+    return model;
+}
+
+module.exports = {
+    setIdpTemplate
+};
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/util/idpUtil/idpUtil.js:125]: Add new empty line

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callPatchGeneralSettingsIdp.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+ 
+         const data = await res.json();
+ 
+         return data;
+     } catch (err) {
+        
+         return null;
+     }
+ }
+
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/identityProvider/callPatchGeneralSettingsIdp.js:41]: It looks like this has not been fixed.

File: b2b-sample/guardio-insurance-sample-application/components/data/templates/enterprise-identity-provider.json [hunk 11]
```diff
@@ -14,9 +14,82 @@
...
+              "isMandatory": true,
+              "isConfidential": false,
+              "options": [],
+              "defaultValue": "",
+              "subProperties": []
+            },
+            {
+              "key": "callbackUrl",
+              "displayName": "Callback Url",
+              "description": "The WSO2 Identity Server URL to which the user needs to be redirected after completing the authentication at the identity provider. The identity provider needs to send the authorization code to this URL upon successful authentication.",
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/data/templates/enterprise-identity-provider.json:79]: Let's make sure a line does not exceed 120 characters.

Comment (lashinijay) [reply b2b-sample/guardio-insurance-sample-application/components/data/templates/enterprise-identity-provider.json:79]: Ref: https://github.com/wso2/samples-is/pull/346#discussion_r951669697

File: b2b-sample/guardio-insurance-sample-application/components/data/templates/facebook.json [hunk 12]
```diff
@@ -104,47 +104,10 @@
...
-              "options": [],
-              "defaultValue": "",
-              "subProperties": []
-            },
             {
               "key": "ClientId",
-              "displayName": "Client Id",
-              "description": "Enter Facebook client identifier value",
+              "displayName": "App ID",
+              "description": "The generated unique ID which is generated when the Facebook OAuth app is created.",
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/data/templates/facebook.json:110]: Suggestion
`The unique ID which is generated when the Facebook OAuth app is created.`

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/buttonGroupIdentityProviderDetails.js [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+import {successTypeDialog, errorTypeDialog} from '../../util/dialog'
+
+export default function ButtonGroupIdentityProviderDetails(props) {
+
+    const { data: session } = useSession();
+    const toaster = useToaster();
+
+    const onIdpDelete = (response) => {
+        if (response) {
+            successTypeDialog(toaster, "Success", "Identity Provider Deleted Successfully");
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/buttonGroupIdentityProviderDetails.js:34]: `Identity provider is deleted successfully`

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identityProviderDetails.js [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+	}, [fetchData]);
+
+	const activeKeyNavSelect = (eventKey) => {
+		setActiveKeyNav(eventKey);
+	}
+
+	const idpDetailsComponent = (activeKey) => {
+		switch (activeKey) {
+			case '1':
+
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identityProviderDetails.js:50]: are these new lines required? If not let's remove them. Check other places too.

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settings.js [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+                }
+            })
+        }
+
+        return errors
+    }
+
+    const onDataSubmit = (response, form) => {
+        if (response) {
+            successTypeDialog(toaster, "Changes Saved Successfully", "Idp updated successfully.");
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settings.js:68]: `Changes are saved successfully`
`Idp is updated successfully`

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js [hunk 16]
```diff
@@ -41,20 +41,19 @@ export default function Settings(props) {
 
     const [activeKeySideNav, setActiveKeySideNav] = useState('1');
 
-    const signOutOnClick = () => orgSignout();
-
     const mainPanelComponenet = (activeKey, session) => {
         switch (activeKey) {
             case '1':
+
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js:47]: are these new lines required?

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/identityProvider/decodeGetDetailedIdentityProvider.js [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+ * under the License.
+ */
+
+import callGetDetailedIdentityProvider from
+    "../../../apiCall/settings/identityProvider/callGetDetailedIdentityProvider";
+
+export default async function decodeGetDetailedIdentityProvider(session, id) {
+
+    try {
+        const res = await callGetDetailedIdentityProvider(session, id);
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/identityProvider/decodeGetDetailedIdentityProvider.js:25]: Can we remove the const res and directly return the value?

Comment (lashinijay) [reply b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/identityProvider/decodeGetDetailedIdentityProvider.js:25]: Check other places too.

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js [hunk 18]
```diff
@@ -19,6 +19,7 @@
  export default function decodeMeObject(user) {
 
     if(user.id==undefined || user.userName==undefined || user.name==undefined || user.emails==undefined ) {
```
[the remark this answers is not in the document]
Comment (lashinijay) [reply b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js:21]: check other places too. Can see the same code snippet in other places too.

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/localResErrors.js [hunk 19]
```diff
@@ -17,14 +17,17 @@
  */
 
 function error404(res, msg) {
+
     return res.status(404).json(msg);
 }
 
 function notPostError(res) {
+
     return error404(res, 'Cannot request data directyly.');
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/localResErrors.js:26]: Check spelling

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js [hunk 20]
```diff
@@ -75,98 +74,39 @@ export default function IdentityProviders() {
         setSelectedTemplate(undefined)
     };
 
-    const onIdPSave = async (formValues, template) => {
-
-        const FIRST_ENTRY = 0;
-        let model = { ...template.idp };
+    const onIdpCreated = (response) => {
+        if (response) {
+            successTypeDialog(toaster, "Success", "Identity Provider Created Successfully");
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js:79]: `Identity Provider is created successfully

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/general.js [hunk 21]
```diff
@@ -0,0 +1,156 @@
...
+
+        return errors;
+    }
+
+    const validate = values => {
+        let errors = {}
+        errors = nameValidate(values.name, errors);
+        errors = descriptionValidate(values.description, errors);
+        
+        return errors
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/general.js:59]: add a `;` in return statement.

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settingsFormSection/settingsFormSelection.js [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+
+export default function SettingsFormSelection(props) {
+
+    const [federatedAuthenticators, setFederatedAuthenticators] = useState(props.federatedAuthenticators);
+    const toaster = useToaster();
+
+    const propList = () => {
+        switch (props.templateId) {
+            case GOOGLE_ID:
+
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settingsFormSection/settingsFormSelection.js:39]: do we need these new lines after the case statement?

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js [hunk 23]
```diff
@@ -41,20 +41,19 @@ export default function Settings(props) {
...
                 return <ViewUserComponent orgName={props.name} orgId={props.orgId} session={session} />;
-            case '2-2':
-                return <AddUserComponent orgName={props.name} orgId={props.orgId} session={session} />;
             case '2-3':
+                
                 return <IdentityProviders orgName={props.name} orgId={props.orgId} session={session} />;
-            case '3-1':
-                return <Application orgName={props.name} session={session} />
+            // case '3-1':
+            //     return <Application orgName={props.name} session={session} />
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js:56]: Shall we remove this if we don't need other than commenting?

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callSwitchOrg.js [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+export default async function callSwitchOrg(subOrgId, accessToken) {
+
+    try {
+        const res = await fetch(
+            `${config.WSO2IS_CLIENT_URL}/api/settings/switchOrg`,
+            getInternalApiRequestOptionsWithParam(null, subOrgId, accessToken)
+        );
+
+        if (res.status != 200) {
+
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callSwitchOrg.js:31]: Do we need these new lines after this if statements? Check other places as well.

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callSwitchOrg.js [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+            return null;
+        }
+
+        const data = await res.json();
+
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callSwitchOrg.js:36]: Shall we remove this new line?

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/getSentDataRequestOptions.js [hunk 26]
```diff
@@ -20,9 +20,10 @@ import config from '../../../config.json';
 
 function sentDataHeader(session) {
     const headers = {
-        "accept": "application/scim+json",
-        "content-type": "application/scim+json",
+        "accept": "application/json",
+        "content-type": "application/json",
         "authorization": "Bearer " + session.accessToken,
+
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/getSentDataRequestOptions.js:26]: Do we need this new line?

File: b2b-sample/guardio-insurance-sample-application/components/data/templates/enterprise-identity-provider.json (hunk 11 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/data/templates/enterprise-identity-provider.json:79]: This is a template file, and IMO it is okay to exceed 120 characters in such files, as we have done the same in our other repos as well.

ex: 
- https://github.com/wso2-enterprise/asgardeo-apps/blob/58c7f91640fae55d4215c33a7075399aeabe7fe2/apps/console/src/extensions/identity-provider-templates/templates/saml/saml.json#L61

File: b2b-sample/guardio-insurance-sample-application/components/data/templates/facebook.json (hunk 12 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/data/templates/facebook.json:110]: The `facebook.json` file has been removed with a later PR,
- https://github.com/wso2/samples-is/blob/master/b2b-sample/libs/business-admin-app/data-access/data-access-common-models-util/src/lib/identityProvider/data/templates

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/general.js (hunk 21 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/general.js:59]: This was after adding the eslint in the `identity-apps` to the project.

### Commit
- https://github.com/wso2/samples-is/commit/e50bfd37f9fdd026c3f685b6fe602f7fdcc542e8#diff-6e806d4538a673d7ea75a914962e3f09e15ecbb1754339a574f54c5c619636d7

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js (hunk 23 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js:56]: This file has been changed with new PR's and these errors has been fixed,

### Commit
- https://github.com/Achintha444/samples-is/commit/3a945da41d8b968ff09ee7268090cc91dd786eb7#diff-de6e61ef1f056a6fd471d69dbf61ddf99deb9ca43519917d6e2dc1f5283551b4

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/getSentDataRequestOptions.js (hunk 26 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/getSentDataRequestOptions.js:26]: Fixed with the PR which moved the functionality to a different file,

### Commit
- https://github.com/Achintha444/samples-is/commit/84e5af312596e01473d17597ddf5641645503597#diff-968532fdc2f70738a9cbab5b80b9bd297eb489f158eea4fb3dce0db7062b4545

File: b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/identityProvider/decodeGetDetailedIdentityProvider.js (hunk 17 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/apiDecode/settings/identityProvider/decodeGetDetailedIdentityProvider.js:25]: Isn't it okay to assign this to variable and return the value ?

File: b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callSwitchOrg.js (hunk 24 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/apiCall/settings/callSwitchOrg.js:31]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settingsFormSection/settingsFormSelection.js (hunk 22 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settingsFormSection/settingsFormSelection.js:39]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js (hunk 20 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/identity-providers.js:79]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js (hunk 18 above)
[the remark this answers is not in the document]
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js:21]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js (hunk 16 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/settings.js:47]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settings.js (hunk 15 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/idpDetailsSections/settings.js:68]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/buttonGroupIdentityProviderDetails.js (hunk 13 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/identity-providers/buttonGroupIdentityProviderDetails.js:34]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/130d957728499e5e0e731eba0b663086f931fe95

**Pull request merged**